cms/djangoapps/contentstore/api/views/course_quality.py

```diff
@@ -103,42 +104,44 @@ def _execute_method_and_log_time(func, *args):
             response = dict(
                 is_self_paced=course.self_paced,
             )
-            if get_bool_param(request, 'sections', all_requested):
-                if course_key_harward:
-                    response.update(
-                        sections=_execute_method_and_log_time(self._sections_quality, course)
-                    )
-                else:
-                    response.update(
-                        sections=self._sections_quality(course)
-                    )
-            if get_bool_param(request, 'subsections', all_requested):
-                if course_key_harward:
-                    response.update(
-                        subsections=_execute_method_and_log_time(self._subsections_quality, course, request)
-                    )
-                else:
-                    response.update(
-                        subsections=self._subsections_quality(course, request)
-                    )
-            if get_bool_param(request, 'units', all_requested):
-                if course_key_harward:
-                    response.update(
-                        units=_execute_method_and_log_time(self._units_quality, course, request)
-                    )
-                else:
-                    response.update(
-                        units=self._units_quality(course, request)
-                    )
-            if get_bool_param(request, 'videos', all_requested):
-                if course_key_harward:
-                    response.update(
-                        videos=_execute_method_and_log_time(self._videos_quality, course)
-                    )
-                else:
-                    response.update(
-                        videos=self._videos_quality(course)
-                    )
+            if not (waffle.switch_is_active('disable_quality_HarvardX+SW12.1x+2016') and
```

Comment:
@asadazam93 If we want to only disable for this particular course, I think we should do either of the following.
1. `if course_key == 'course-v1:HarvardX+SW12.1x+2016')` check is sufficient. I do not find anything interesting in adding this waffle flag to make an additional call to db where the same behavior can be achieved via only one check on `course_key`.

2. If we want to add this check for multiple courses, then I guess it makes sense to change the waffle flag to be a `CourseWaffleFlag` and add course keys as we came to know.

Going to tag @fysheets for more thoughts which I might be missing.

Reply:
I think these two options sound good. Using option number two will prevent us from making another code change in the interim if additional courses are found to be causing issues as well. I would lean towards option 2 if it's not a much bigger lift.

Reply:
option 2 seems to be a good one. I have made the changes accordingly